### PR TITLE
bump libocr; add context

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-2887

Requires:
- https://github.com/smartcontractkit/chainlink-common/pull/490

Supports:
- https://github.com/smartcontractkit/chainlink/pull/14011

core ref: 045c76f80d7249d572353ae904acc640b0597eef

### CODE REVIEW

File: commit/merkleroot/observation.go [hunk 1]
```diff
@@ -29,9 +31,12 @@ import (
...
 )
 
-func (w *Processor) ObservationQuorum(_ ocr3types.OutcomeContext, _ types.Query) (ocr3types.Quorum, error) {
+func (w *Processor) ObservationQuorum(
+	_ context.Context, _ ocr3types.OutcomeContext, _ types.Query, aos []types.AttributedObservation,
+) (bool, error) {
 	// Across all chains we require at least 2F+1 observations.
-	return ocr3types.QuorumTwoFPlusOne, nil
+	return quorumhelper.ObservationCountReachesObservationQuorum(
+		quorumhelper.QuorumTwoFPlusOne, w.reportingCfg.N, w.reportingCfg.F, aos), nil
```
[the remark this answers is not in the document]
Comment (jmank88) [reply commit/merkleroot/observation.go:39]: Total number of nodes

[the remark this answers is not in the document]
Comment (jmank88) [reply commit/merkleroot/observation.go:39]: It is not actually used in the implementation for the `QuorumTwoFPlusOne` case, but we aren't supposed to depend on that as a caller, so passing for completeness and forward compatability.

File: execute/plugin.go [hunk 2]
```diff
@@ -243,15 +249,17 @@ func (p *Plugin) Reports(seqNr uint64, outcome ocr3types.Outcome) ([]ocr3types.R
 	}
 
 	// TODO: this function should be pure, a context should not be needed.
```
Comment (winder) [inline execute/plugin.go:251]:
```suggestion
```
You can also remove a similar TODO in `execute/report/builder.go`

File: commit/merkleroot/observation.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (winder) [reply commit/merkleroot/observation.go:39]: Makes sense, thanks.

File: internal/plugincommon/ccipreader.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
+package plugincommon
```
Comment (winder) [inline internal/plugincommon/ccipreader.go:1]: Delete this file, it's no longer used.

File: internal/plugincommon/discovery/processor_test.go [hunk 4]
```diff
@@ -6,13 +6,12 @@ import (
 	"testing"
 
 	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/libocr/commontypes"
+	ragep2ptypes "github.com/smartcontractkit/libocr/ragep2p/types"
```
Comment (winder) [inline internal/plugincommon/discovery/processor_test.go:10]: nit: these got moved into the internal dep section

File: commit/factory.go [hunk 5]
```diff
@@ -90,7 +90,7 @@ func NewPluginFactory(
 	}
 }
 
-func (p *PluginFactory) NewReportingPlugin(config ocr3types.ReportingPluginConfig,
+func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types.ReportingPluginConfig,
```
Comment (winder) [inline commit/factory.go:93]: What is this context? Is there a proper lifecycle context now?

This is called by `chainlink` code, so the full integration test will break as a result.

File: pkg/reader/usdc_reader.go [hunk 6]
```diff
@@ -74,7 +74,7 @@ func (m MessageSentEvent) unpackID() (eventID, error) {
 	return result, nil
 }
 
-func NewUSDCMessageReader(
+func NewUSDCMessageReader(ctx context.Context,
 	lggr logger.Logger,
```
Comment (winder) [inline pkg/reader/usdc_reader.go:78]:
```suggestion
func NewUSDCMessageReader(
	ctx context.Context,
	lggr logger.Logger,
```

File: execute/plugin.go (hunk 2 above)
Comment (winder) [reply execute/plugin.go]: Nice, thanks

File: commit/plugin.go [hunk 7]
```diff
@@ -176,9 +176,12 @@ func (p *Plugin) Query(ctx context.Context, outCtx ocr3types.OutcomeContext) (ty
...
 }
 
-func (p *Plugin) ObservationQuorum(_ ocr3types.OutcomeContext, _ types.Query) (ocr3types.Quorum, error) {
+func (p *Plugin) ObservationQuorum(
+	ctx context.Context, _ ocr3types.OutcomeContext, _ types.Query, aos []types.AttributedObservation,
+) (bool, error) {
 	// Across all chains we require at least 2F+1 observations.
-	return ocr3types.QuorumTwoFPlusOne, nil
+	return quorumhelper.ObservationCountReachesObservationQuorum(
+		quorumhelper.QuorumFPlusOne, p.reportingCfg.N, p.reportingCfg.F, aos), nil
```
Comment (makramkd) [inline commit/plugin.go:184]:
```suggestion
	return quorumhelper.ObservationCountReachesObservationQuorum(
		quorumhelper.QuorumTwoFPlusOne, p.reportingCfg.N, p.reportingCfg.F, aos), nil
```

**Review by makramkd — CHANGES_REQUESTED** [review]

File: execute/tokendata/usdc/http_test.go [hunk 8]
```diff
@@ -64,7 +65,7 @@ func Test_NewHTTPClient_New(t *testing.T) {
 func Test_HTTPClient_Get(t *testing.T) {
 	tt := []struct {
 		name               string
-		getTs              func() *httptest.Server
+		getTs              func(t *testing.T) *httptest.Server
```
Comment (makramkd) [inline execute/tokendata/usdc/http_test.go:68]: Why this change? I don't see the `testing.T` used anywhere in these funcs.

File: go.mod [hunk 9]
```diff
@@ -5,8 +5,8 @@ go 1.22.5
 require (
 	github.com/deckarep/golang-set/v2 v2.6.0
 	github.com/smartcontractkit/chain-selectors v1.0.23
-	github.com/smartcontractkit/chainlink-common v0.2.3-0.20241008175210-167715aa8613
-	github.com/smartcontractkit/libocr v0.0.0-20240717100443-f6226e09bee7
+	github.com/smartcontractkit/chainlink-common v0.3.0
```
Comment (makramkd) [inline go.mod:8]: Noice, semver!

File: commit/plugin.go (hunk 7 above)
Comment (jmank88) [reply commit/plugin.go:184]: Ah, good catch

File: execute/tokendata/usdc/http_test.go (hunk 8 above)
Comment (jmank88) [reply execute/tokendata/usdc/http_test.go:68]: The `default timeout` case uses a `require.NoError`

Comment (makramkd) [reply execute/tokendata/usdc/http_test.go:68]: Oh I see, they use the top level `t *testing.T` in the test func. I usually rename nested `testing.T` to `tt` but thats just preference.

**Review by makramkd — APPROVED** [review]

**Comment by makramkd** [comment]
Integration test will fail until https://github.com/smartcontractkit/chainlink/pull/14011 is merged

File: execute/tokendata/usdc/http_test.go (hunk 8 above)
Comment (jmank88) [reply execute/tokendata/usdc/http_test.go:68]: Hmmm I think using a different name could be error prone, because the original `t` is still in scope to be referenced accidentally. This happens with ctx and err often as well.

**Comment by jmank88** [comment]
> Integration test will fail until [smartcontractkit/chainlink#14011](https://github.com/smartcontractkit/chainlink/pull/14011) is merged

Is there a way to configure it to run against latest core like we do in the non-evm repos?
image
